quickstart-python/services/monday_api.py

```diff
@@ -0,0 +1,41 @@
+from typing import Callable
+
+import monday_code
+from monday_code.exceptions import NotFoundException, ApiException
+from urllib3.exceptions import RequestError
+
+from errors import MondayCodeAPIError
+from handlers import mcode_configuration
+from models import APITypes
+
+
+def api_instance_factory(api_type: APITypes, api_client):
+    """Factory method to create an instance of the requested API type"""
+    if api_type == APITypes.QUEUE:
+        return monday_code.QueueApi(api_client)
+    elif api_type == APITypes.SECURE_STORAGE:
+        return monday_code.SecureStorageApi(api_client)
+    elif api_type == APITypes.SECRETS:
+        return monday_code.SecretApi(api_client)
+    else:
+        raise MondayCodeAPIError(f"Invalid API type", api_type)
+
+
+def with_monday_api(api_type: APITypes, method_name: str):
```

Comment:
what is the added value of this decorator?

Reply:
This decorator makes it easier to use the monday-api, provides the API instance to the decorated function and handles all the errors and logging.

I've updated it in the latest merge, also implemented usage of it in every call.